ui/app/src/model/datasource-api.ts

```diff
@@ -11,45 +11,203 @@
 import { DatasourceApi } from '@perses-dev/dashboards';
+import LRUCache from 'lru-cache';
 import { fetchDatasourceList, fetchGlobalDatasourceList } from './datasource-client';
 
-export function useDatasourceApi(): DatasourceApi {
-  return {
-    getDatasource: async (project, selector) => {
-      return fetchDatasourceList(project, selector.kind, selector.name ? undefined : true, selector.name).then(
-        (list) => {
-          // hopefully it should return at most one element
-          if (list[0] !== undefined) {
-            return {
-              resource: list[0],
-              proxyUrl: getProxyUrl(list[0]),
-            };
-          }
-        }
-      );
-    },
-    getGlobalDatasource: async (selector) => {
-      return fetchGlobalDatasourceList(selector.kind, selector.name ? undefined : true, selector.name).then((list) => {
-        // hopefully it should return at most one element
-        if (list[0] !== undefined) {
-          return {
-            resource: list[0],
-            proxyUrl: getProxyUrl(list[0]),
-          };
-        }
-      });
-    },
-
-    listDatasources: async (project, pluginKind) => {
-      return fetchDatasourceList(project, pluginKind);
-    },
-
-    listGlobalDatasources: async (pluginKind) => {
-      return fetchGlobalDatasourceList(pluginKind);
-    },
-  };
+export class HTTPDatasourceAPI implements DatasourceApi {
+  getDatasource(
+    project: string,
+    selector: DatasourceSelector
+  ): Promise<{ resource: Datasource; proxyUrl: string } | undefined> {
+    return fetchDatasourceList(project, selector.kind, selector.name ? undefined : true, selector.name).then((list) => {
+      // hopefully it should return at most one element
+      if (list[0] !== undefined) {
+        return {
+          resource: list[0],
+          proxyUrl: getProxyUrl(list[0]),
+        };
+      }
+    });
+  }
+
+  getGlobalDatasource(
+    selector: DatasourceSelector
+  ): Promise<{ resource: GlobalDatasource; proxyUrl: string } | undefined> {
+    return fetchGlobalDatasourceList(selector.kind, selector.name ? undefined : true, selector.name).then((list) => {
+      // hopefully it should return at most one element
+      if (list[0] !== undefined) {
+        return {
+          resource: list[0],
+          proxyUrl: getProxyUrl(list[0]),
+        };
+      }
+    });
+  }
+
+  listDatasources(project: string, pluginKind: string | undefined): Promise<Datasource[]> {
+    return fetchDatasourceList(project, pluginKind);
+  }
+
+  listGlobalDatasources(pluginKind: string | undefined): Promise<GlobalDatasource[]> {
+    return fetchGlobalDatasourceList(pluginKind);
+  }
+}
+
+class Cache {
+  private datasources: LRUCache<string, Datasource>;
+  private emptyDatasources: LRUCache<string, boolean>;
+  private globalDatasources: LRUCache<string, GlobalDatasource>;
+
+  constructor() {
+    const option = { ttl: 5 * 60 * 1000, ttlAutopurge: true };
```

Comment:
Can you add a quick inline comment about what `ttl` is / why these values were picked?